test/unit/pusher_test.rb

```diff
@@ -1,13 +1,88 @@
 require 'test_helper'
 
 class PusherTest < ActiveSupport::TestCase
-  context "creating a new gemcutter" do
-    setup do
-      @user = create(:user, email: "user@example.com")
-      @gem = gem_file
-      @cutter = Pusher.new(@user, @gem)
+  setup do
+    @user = create(:user, email: "user@example.com")
+    @gem = gem_file
+    @cutter = Pusher.new(@user, @gem)
+  end
+
+  context "processing incoming gems" do
+    should "work normally when things go well" do
+      @cutter.stubs(:pull_spec).returns true
+      @cutter.stubs(:find).returns true
+      @cutter.stubs(:validate_gem_version).returns true
+      @cutter.stubs(:authorize).returns true
+      @cutter.stubs(:validate).returns true
+      @cutter.stubs(:save)
+
+      legit_gem = create(:rubygem, name: 'legit-gem')
+      @cutter.stubs(:rubygem).returns(legit_gem)
+
+      @cutter.process
+    end
+
+    should "not attempt to find rubygem if spec can't be pulled" do
+      @cutter.stubs(:pull_spec).returns false
```

Comment:
(Grammar, trivial note.) The should text description uses the word `rubygem` and not `gem`. The other should description texts use `gem`. Can that be harmonized?